oauthproxy.go

```diff
@@ -118,6 +123,7 @@ func NewOAuthProxy(opts *options.Options, validator func(string) bool) (*OAuthPr
 	if err != nil {
 		return nil, fmt.Errorf("error intiailising provider: %v", err)
 	}
+	provider.Data().CodeChallengeMethod = providers.ParseCodeChallengeMethod(opts)
```

Comment:
Why is this not set up as part of the construction of the provider data itself, seems a weird place to override it here

Reply:
Moved!